src/Fl_Text_Display.cxx

```diff
@@ -2293,19 +2301,27 @@ void Fl_Text_Display::draw_string(int style,
     styleRec = mStyleTable + si;
     font  = styleRec->font;
     fsize = styleRec->size;
+    bgbasecolor = (styleRec->attr&ATTR_BGCOLOR) ? styleRec->bgcolor : color();
 
     if (style & PRIMARY_MASK) {
       if (Fl::focus() == (Fl_Widget*)this) {
-        if (Fl::screen_driver()->has_marked_text() && Fl::compose_state)
-          background = color();// Mac OS: underline marked text
-        else
+        if (Fl::screen_driver()->has_marked_text() && Fl::compose_state) {
+          background = bgbasecolor; // Mac OS: underline marked text
+        } else {
           background = selection_color();
```

Comment:
I like the additional curly braces! This makes the nested if/else constructs better readable.